pkg/workloads/cortex/lib/util.py

```diff
@@ -297,3 +297,7 @@ def render_jinja_template(jinja_template_file: str, context: dict) -> str:
 
     template = env.get_template(str(template_path.name))
     return template.render(**context)
+
+
+def available_aws_credentials() -> bool:
+    return os.getenv("AWS_ACCESS_KEY_ID") and os.getenv("AWS_SECRET_ACCESS_KEY")
```

Comment:
Would it be risky to check this in the S3 constructor instead of checking it outside? For example, to leave the calls like `s3_client = S3(bucket_name)`, but in the constructor, set the unsigned config if the creds aren't available? Or could something break if we do it that way?

Reply:
Moved the process of checking the availability of AWS credentials inside the S3 class' constructor.